CMake/mlpack.cmake

```diff
@@ -528,11 +528,15 @@ endmacro()
 macro(find_mlpack_internal)
 
   set(CURRENT_PATH ${ARGN})
+
   if (CURRENT_PATH)
-    find_path(MLPACK_INCLUDE_DIR
-      NAMES mlpack.hpp
-      PATHS "${CURRENT_PATH}/deps/mlpack/src/include" "${CMAKE_CURRENT_SOURCE_DIR}/src/"
-      NO_DEFAULT_PATH)
+    file(GLOB MLPACK_SEARCH_PATHS
+      ${CMAKE_BINARY_DIR}/deps/mlpack-[0-9]*.[0-9]*.[0-9]*/src/)
+
+    if (EXISTS ${MLPACK_SEARCH_PATHS}/mlpack.hpp)
```

Comment:
Does this assume that `MLPACK_SEARCH_PATHS` is only one path?  I know that `GLOB` can return a list, so maybe it is best to take the last element out of the list (assuming that it is ordered, you would have to check), so that we use the most recent version of mlpack.

Reply:
when I checked inside it it was only one path, this is the reason I used it in here. 
Technically it should not return more than one because we are specifying to look in the `CMAKE_BINARY_PATH`

Reply:
What if the user has a build directory that had an older version of mlpack already in `deps/`?  Then they will get multiple directories.